atest/testdata/standard_libraries/builtin/wait_until_keyword_succeeds.robot

```diff
@@ -115,3 +125,13 @@ Wait Until Inside User Keyword
 Timeouted UK with Wait Until KW
     [Timeout]    ${timeout}
     Wait Until Keyword Succeeds    100ms    10ms    Fail    Error in timeouted UK
+
+Get Length On Returned Variable Works On Last Attempt
+    ${myvar} =    return_value_after_retries    Hello
+    Get Length    ${myvar}
+    Fail    Test PASSES : variable is no more None
```

Comment:
This keyword doesn't test what it is supposed to test. The variable `${myvar}` exists every time, so accessing it never causes a variable related failure.

You have two good ways to actually test that WUKS first fails when a variable doesn't exist and then succeeds when it does:

1. Just use a variable that doesn't exist. Verify that WUKS was executed multiple times on the test execution side.

2. Dynamically create a variable after first accessing it when it didn't yet exist. The keyword that is used with WUKS could be something like this:

    ```robotframework
    *** Keywords ***
    Example
        Log    ${dynamically set variable}    # doesn't exist when this keyword is executed first time
        Set Test Variable    ${dynamically set variable}    value
    ```

    Also in this case I'd like to verify on the execution side that the first round really failed.